docs/services-web-app.md

```diff
@@ -12,3 +29,255 @@ access to private repos and increases API quotas. To set up a personal access to
+
+The app uses run-mode to make decisions that are environment-dependant (cookies, next-server...).
+This package configuration is also needed for proper functionality of the Auth Package.
+
+Make sure required environment variables are configured for proper functionality of this package.
+(See [run-mode package](./packages-api.md#run-mode) for details on how to set up the run-mode
+package)
+
+_Note_: this package requires different values depending on environemnt, make sure to update these
+when running on dev/test/production mode
+
+### Auth
+
+This service depends on the auth package; Make sure required environment variables are configured
+for proper functionality of this package. (See [auth package](./packages-api.md#auth) for details on
+how to set up the auth package)
+
+_Note_: this package requires different environment vairables and values depending on environemnt,
+make sure to update these when running on dev/test/production mode
+
+## Adding Local Certificates
+
+The application must run on https for IBMid authentication to work properly. For such purposes,
+local certificates must be generated to run on development environment. You may use
+[mkcert](https://github.com/FiloSottile/mkcert#installation) tool for this. With the tool
+downloaded, the certificates will be automatically generated next time you
+[run the app securely](#running-app-securely)
+
+## Running App Securely
+
+To run the app on https, it must be served from [server.js](../services/web-app/server.js). To do
+this, run the following command from the [web-app's directory](../services/web-app):
+
+`npm run start-secure`
+
+## Protecting a Route
+
+### Protecting Server-Side Rendered Pages
+
+In order to require authentication before a user can access a server-side rendered page use the
+`getPropsWithAuth` utlity function to wrap the pages' _getServerSideProps_ function and wrap the
+pages' content inside the `RequireAuth` component.
+
+#### getPropsWithAuth
+
+`getPropsWithAuth` expects an `authorizationChecker` function that receives the server side context
+as a param and must return a boolean value indicating whether the user is authorized to view the
+content or not:
+
+```
+import { getPropsWithAuth } from '@/utils/getPropsWithAuth'
+import { retrieveUser } from '@/utils/retrieveUser'
+...
+// Your custom authorization logic here, this one considers the user as authorized if it's email address ends in "ibm.com"
+const isValidIbmUser = async (context) => {
+  const user = await retrieveUser(context)
+  if (user) {
+    return user.email?.endsWith('ibm.com')
+  }
+  return false
+}
+...
+export const getServerSideProps = getPropsWithAuth(isValidIbmUser, async (/* context */) => {
+  // Your normal `getServerSideProps` code here
+  return {
+    props: {
+      // Your custom props here
+    }
+  }
+})
+```
+
+Some `authorizationChecker` functions that serve common purposes are exported from
+[utils/auth-checkers](..services/web-app/utils/auth-checkers):
```

Comment:
```suggestion
`services/web-app/utils/auth-checkers`:
```
Since this is out of band of the docs, it's probably best not to link to it directly